src/classes/PromoModals.js

```diff
@@ -33,23 +35,51 @@ class PromoModals {
 
 	static isTimeForAPlusPromo() { return this._isTimeForAPromo(PLUS); }
 
+	static isTimeForInsightsPromo() { return this._isTimeForAPromo(INSIGHTS); }
+
 	static recordPlusPromoSighting() { this._recordPromoSighting(PLUS); }
 
-	// TODO integrate the Insights promo modal into the "has it been long enough since last modal?" logic here
+	static recordInsightsPromoSighting() { this._recordPromoSighting(INSIGHTS); }
+
 	static _isTimeForAPromo(type) {
-		const lastSeenTime = conf[`${type}_${PROMO_MODAL_LAST_SEEN}`];
+		const lastSeenPlusPromo = conf[`${PLUS}_${PROMO_MODAL_LAST_SEEN}`];
+		const lastSeenInsightsPromo = conf[`${INSIGHTS}_${PROMO_MODAL_LAST_SEEN}`];
+		const lastSeenPromo = Math.max(lastSeenPlusPromo, lastSeenInsightsPromo);
 
-		if (lastSeenTime === null) { return true; }
+		if (conf.notify_promotions === false) { return false; }
```

Comment:
Might as well move this early return to the top of the function

Reply:
Moving the early return to the top!